Sources/CacheAdvance/BigEndianHostSwappable.swift

```diff
@@ -31,10 +31,10 @@ extension BigEndianHostSwappable {
     ///
     /// - Parameter data: A data blob representing encodable data. Must be of length `Self.storageLength`.
     init(_ data: Data) {
-        let decodedSize = withUnsafePointer(to: data) {
```

Comment:
I don't have enough context to review these changes but everything else LGTM!

Reply:
I am confident in these changes (we have _excellent_ test coverage), so I'm okay merging this in.

What's happening in this method is we're instantiating a `BigEndianHostSwappable` (i.e. a `UInt64`, `UInt32`, or `UInt8`) from a `Data` blob.

The way we did that was:
1. Get a pointer to the underlying data (`withUnsafePointer(to: data) { ... }`)
2. Turn that pointer into a buffer pointer (i.e. a pointer range) starting at the pointer and ending at the length of storage required to store this type (`UnsafeRawBufferPointer(start: $0, count: Self.storageLength)`)
3. Using that buffer pointer to `load` the underlying `BigEndianHostSwappable`, i.e. read the pointer buffer from memory and cast it as the current type. (`decodedSize.load(as: Self.self)`)
4. Swap the read-in value to have the same Endian-ness as the current machine – this is only necessary if the file has been transferred between machines of different Endian-ness. (`Self.swapToHost(...)`)

The way we're doing it now is:
1. Get the underlying buffer pointer for the underlying data (`withUnsafeBytes { ... }`)
2. Using that buffer pointer (i.e. a pointer range) to `load` the underlying `BigEndianHostSwappable`, i.e. read the pointer buffer from memory and cast it as the current type. (`$0.load(as: Self.self)`)
3. Swap the read-in value to have the same Endian-ness as the current machine – this is only necessary if the file has been transferred between machines of different Endian-ness. (`Self.swapToHost(...)`)

The issue with the old code (which of course I figured out by typing all of this out) was that we were working over the unsafe buffer pointer range _outside_ of the `withUnsafePointer` call. If I change this code to read:
```
        let decodedSize = withUnsafePointer(to: data) {
            return UnsafeRawBufferPointer(start: $0, count: Self.storageLength).load(as: Self.self)
        }
```
The tests pass in Release! So changing from `withUnsafePointer` to `withUnsafeBytes` wasn't the issue. Instead, it was where we were interacting with the buffer! That said, I do like the semantics of `withUnsafeBytes` better.

Reply:
To make sure I follow, the issue was that we were violating this requirement?
<img width="1000" alt="Screen Shot 2022-09-02 at 6 47 19 PM" src="https://user-images.githubusercontent.com/1791049/188251231-147e4f29-bad0-46db-a34f-84b756bf241b.png">

Reply:
Exactly!